File: hard/Number of Ways to Paint N × 3 Grid/C++/main.cpp
// Problem: Number of Ways to Paint N × 3 Grid
// Link to the problem: https://leetcode.com/problems/number-of-ways-to-paint-n-3-grid/
class Solution
{
public:
    int numOfWays(int n)
    {
        const long long mod = 1000000007;
        long long a = 6, b = 6;
        for (int i = 1; i < n; i++)
        {
            const long long x = ((3 * a) + (2 * b)) % mod;
            const long long y = ((2 * a) + (2 * b)) % mod;
            a = x;
            b = y;
        }
        const int ans = (a + b) % mod;
        return ans;
    }
};

File: hard/Number of Ways to Paint N × 3 Grid/Rust/main.rs
// Problem: Number of Ways to Paint N × 3 Grid
// Link to the problem: https://leetcode.com/problems/number-of-ways-to-paint-n-3-grid/
impl Solution {
    pub fn num_of_ways(n: i32) -> i32 {
        let MOD: i64 = 1000000007;
        let mut a: i64 = 6;
        let mut b: i64 = 6;
        for i in 1..n {
            let x: i64 = ((3 * a) + (2 * b)) % MOD;
            let y: i64 = ((2 * a) + (2 * b)) % MOD;
            a = x;
            b = y;
        }
        let ans: i32 = ((a + b) % MOD) as i32;
        return ans;
    }
}
